### PR TITLE
Add Counter

### PR DESCRIPTION
**Pull request by 9il**

### CODE REVIEW

File: source/mir/range.d [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+Data size counter.
+
+Does not store anything
++/
+struct Counter(T)
+{
+    import std.range: isInputRange, ElementType;
+    import std.traits: isImplicitlyConvertible, isSomeChar;
+    ///
+    size_t _count;
```
Comment (wilzbach) [inline source/mir/range.d]: Why do you expose this?

Comment (9il) [reply source/mir/range.d]: To make hacks easily. This looks useful.

**Comment by 9il** [comment]
> How about overloading the ~= assignment operator for this?

~= is array / appender like API. I think it more about data, but counter does not store the data.

**Comment by 9il** [comment]
> In the numir Counter discussions, you had asked for more unittests. I see two in this, just testing for Counter!char and Counter!wchar, but not dchar.

It already covers algorithm logic.



> Also, Appender requires a dynamic array, which is why you commonly see Appender!string and not Appender!char. Might make sense to add tests for Counter!string/Counter!wstring/Counter!dstring.

`Counter!string` corresponds to `Appender!(string[])`

**Pull request merged**